astropy/coordinates/sky_coordinate.py

```diff
@@ -1872,13 +1872,25 @@ def guess_from_table(cls, table, **coord_kwargs):
         -------
         newsc : same as this class
             The new `SkyCoord` (or subclass) object.
+
+        Raises
+        ------
+        ValueError
+            If more than one match is found in the table for a component,
+            unless the additional matches are also valid frame component names.
+            If a "coord_kwargs" is provided for value also found in the table.
+
         """
         _frame_cls, _frame_kwargs = _get_frame_without_data([], coord_kwargs)
         frame = _frame_cls(**_frame_kwargs)
         coord_kwargs['frame'] = coord_kwargs.get('frame', frame)
 
+        representation_component_names = tuple({
+            **frame.get_representation_component_names(), **frame.get_representation_component_names("s")
```

Comment:
Nitpick, but it seems likely that it is faster to just do
```
= tuple(frame.get_representation_component_names()) + tuple(frame.get_representation_component_names("s"))
```

Reply:
agreed. I'm changing over all relevant lists to sets.